src/library/scala/collection/GenTraversableOnce.scala

```diff
@@ -80,6 +94,9 @@ trait GenTraversableOnce[+A] extends Any {
   def size: Int
 
   /** Tests whether the $coll is empty.
+    *
+    * Note: Implementations in subclasses that are not repeatedly traversable must take
```

Comment:
Wouldn't it be better to phrase this as a guarantee to the consumer of the API that `isEmpty` does not traverse a collection or consume any elements (so may freely be used on a TraversableOnce)?